code.user.js

```diff
@@ -29,6 +29,29 @@
 (function($, async) {
     $.noConflict(true);
 
+    var ChangeHashName = {
```

Comment:
Not a fan of hard coding all possible cards, can this be done differently?

Reply:
> Not a fan of hard coding all possible cards, can this be done differently?


The following is a script I wrote that can retrieve the correct HashName for all items in a Steam game market. 

```
// ==UserScript==
// @name         GetAllCorrectHashNameItems
// @version      1.0
// @description  GetAllCorrectHashNameItems
// @author       第三异星
// @match        https://steamcommunity.com//tradingcards/boostercreator/
// @icon         https://www.google.com/s2/favicons?sz=64&domain=steamcommunity.com
// @require      https://code.jquery.com/jquery-3.3.1.min.js
// ==/UserScript==

(async function ($) {
    $.noConflict(true);
    let button = document.createElement('button');
    button.textContent = 'GetAllCorrectHashNameItems';
    button.style.position = 'fixed';
    button.style.top = '50px';
    button.style.left = '10px';
    button.style.zIndex = 9999;
    document.body.appendChild(button);

    button.addEventListener('click', async function () {
        let url = 'https://steamcommunity.com/market/search/render/?start=0&count=100&q=&category_753_Event%5B%5D=any&category_753_Game%5B%5D=tag_app_1186460&category_753_item_class%5B%5D=tag_item_class_2&category_753_item_class%5B%5D=tag_item_class_4&category_753_item_class%5B%5D=tag_item_class_3&category_753_item_class%5B%5D=tag_item_class_5&category_753_item_class%5B%5D=tag_item_class_10&category_753_item_class%5B%5D=tag_item_class_6&appid=753'
        await $.get(url).then(allItems => {
            console.log(allItems)
            let parser = new DOMParser();
            let doc = parser.parseFromString(allItems.results_html, 'text/html');
            let elements = doc.querySelectorAll('.market_listing_row.market_recent_listing_row.market_listing_searchresult');
            elements.forEach((element) => {
                let CorrectHashName = element.getAttribute('data-hash-name');
                console.log(CorrectHashName)
            });
        });
    });


})(jQuery);
```

However, it cannot be used with SEE. SEE’s getItemOrdersHistogram function will call getMarketItemNameId, and getMarketItemNameId requires the correct HashName to obtain the correct MarketItemNameId. If the obtained MarketItemNameId is incorrect, it will result in histogram == null. This will cause the calculatedPrice in the function calculateSellPriceBeforeFees(history, histogram, applyOffset, minPriceBeforeFees, maxPriceBeforeFees) to be equal to maxPriceBeforeFees.


The following is the g_rgAssets obtained from Steam through SEE. You can see that the market_hash_name: 1186460-Zang Guang (Trading Card) is incorrect. The correct market_hash_name is 1186460-Zang Guang. However, although we can read the g_rgAssets from Steam through SEE, we cannot determine whether the market_hash_name of g_rgAssets is correct. Therefore, it is impossible for us to use other methods to correct the wrong HashName.

```
var g_rgAssets = {
    "753": {
        "6": {
            "28214282451": {
                "currency": 0,
                "appid": 753,
                "contextid": "6",
                "id": "28214282451",
                "classid": "4005818963",
                "instanceid": "0",
                "amount": "1",
                "status": 2,
                "original_amount": "1",
                "unowned_id": "28287344309",
                "unowned_contextid": "6",
                "background_color": "",
                "icon_url": "IzMF03bk9WpSBq-S-ekoE33L-iLqGFHVaU25ZzQNQcXdA3g5gMEPvUZZEfSaKqhBT8kyvCOETZfYgdQOxXsKnDgGbDbxyXhoOKBJYI6el0Da9qeoSTHVMXLmCWfQEV9vGL4NN2vb-zCht77AFzHARLopEFsEePAN9zVOPsvbbBc-1YQMqjDpxRctUAYmdYNEfx2_z2YsOLAkzyRPc81VkCT2csGMgg40Pk5uW7HnX-iQbdSjlnlwX05uSaESY4qXuiLr_omxNKCQa_U1AuK7R-A",
                "icon_url_large": "IzMF03bk9WpSBq-S-ekoE33L-iLqGFHVaU25ZzQNQcXdA3g5gMEPvUZZEfSaKqhBT8kyvCOETZfYgdQOxXsKnDgGbDbxyXhoOKBJYI6el0Da9qeoSTHVMXLmCWfQEV9vGL4NN2vb-zCht77AFzHARLopEFsEePAN9zVOPsvbbBc-1YQMqjDpxRctUAYmdYNEfx2_z2YsOLAkzyRPc81VkCT2csGMgg40Pk5uW7HnX-iQbdSjlnlwX05uSaESY4qXuiLr_omxNKCQa_U1AuK7R-A",
                "descriptions": [{"value": "\u4e0d\u53ef\u63a7\u89d2\u8272\uff1a\u81e7\u5e7f\u3002\u6b66\u5668\uff1a\u547d\u60ac\u4e00\u7ebf\uff0c\u9053\uff1a\u6df7\u4e71\u3002"}],
                "tradable": 1,
                "owner_actions": [{
                    "link": "https:\/\/steamcommunity.com\/my\/gamecards\/1186460\/",
                    "name": "\u67e5\u770b\u5fbd\u7ae0\u8fdb\u5ea6"
                }, {
                    "link": "javascript:GetGooValue( '%contextid%', '%assetid%', 1186460, 11, 0 )",
                    "name": "\u8f6c\u6362\u6210\u5b9d\u77f3\u2026"
                }],
                "name": "\u81e7\u5e7f (\u96c6\u6362\u5f0f\u5361\u724c)",
                "type": "\u4fa0\u9053\u6e38\u6b4c \/ Songs Of Wuxia \u96c6\u6362\u5f0f\u5361\u724c",
                "market_name": "\u81e7\u5e7f (\u96c6\u6362\u5f0f\u5361\u724c)",
                "market_hash_name": "1186460-Zang Guang (Trading Card)",
                "market_fee_app": 1186460,
                "commodity": 1,
                "market_tradable_restriction": 7,
                "market_marketable_restriction": 7,
                "marketable": 1,
                "app_icon": "https:\/\/cdn.akamai.steamstatic.com\/steamcommunity\/public\/images\/apps\/753\/1d0167575d746dadea7706685c0f3c01c8aeb6d8.jpg",
                "owner": 0
            }
        }
    }
};


```